pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -215,7 +215,12 @@ nixBuild() {
 }
 
 nixFlakeBuild() {
-    if [ -z "$buildHost" ]; then
+    if [[ -z "$buildHost" && -z "$targetHost" ]] &&
+       ! [ "$action" = switch -o "$action" = boot ]
+    then
+        nix build "$@" &&\
```

Comment:
I added `&&\` here in the last force-push.
This fixes nixos-rebuild switching the system to a preexisting `./result` symlink on build failure.

Reply:
@jonringer That's right. Fixed!